eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/runner/ClusteringCommandLineRunnerTest.java

```diff
@@ -0,0 +1,159 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.batch.core.BatchStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobExecutionException;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.batch.test.JobRepositoryTestUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.eva.accession.clustering.parameters.InputParameters;
+
+import javax.sql.DataSource;
+import java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static uk.ac.ebi.eva.accession.clustering.runner.RunnerTestConfiguration.TEST_JOB_NAME;
+import static uk.ac.ebi.eva.accession.clustering.runner.RunnerTestConfiguration.TEST_STEP_1_NAME;
+import static uk.ac.ebi.eva.accession.clustering.runner.RunnerTestConfiguration.TEST_STEP_2_NAME;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes={RunnerTestConfiguration.class})
+@TestPropertySource("classpath:clustering-pipeline-test.properties")
+public class ClusteringCommandLineRunnerTest {
+
+    @Autowired
+    private InputParameters inputParameters;
+
+    @Autowired
+    private JobRepository jobRepository;
+
+    @Autowired
+    private JobExplorer jobExplorer;
+
+    @Autowired
+    private DataSource datasource;
+
+    @Autowired
+    private ClusteringCommandLineRunner runner;
+
+    private JobRepositoryTestUtils jobRepositoryTestUtils;
+
+    @Before
+    public void setUp() {
+        jobRepositoryTestUtils = new JobRepositoryTestUtils(jobRepository, datasource);
+        runner.setJobNames(TEST_JOB_NAME);
+    }
+
+    @After
+    public void tearDown() {
+        jobRepositoryTestUtils.removeJobExecutions();
+        inputParameters.setForceRestart(false);
+    }
+
+    @Test
+    public void runJobWithNoErrors() throws JobExecutionException {
+        runner.run();
+        assertEquals(ClusteringCommandLineRunner.EXIT_WITHOUT_ERRORS, runner.getExitCode());
+    }
+
+    @Test
+    public void runJobWithNoName() throws JobExecutionException {
+        runner.setJobNames(null);
+        runner.run();
+        assertEquals(ClusteringCommandLineRunner.EXIT_WITH_ERRORS, runner.getExitCode());
+    }
+
+    @Test
+    public void runNonExistentJob() throws JobExecutionException {
+        runner.setJobNames("NOT_EXISTENT_JOB");
+        runner.run();
+        assertEquals(ClusteringCommandLineRunner.EXIT_WITH_ERRORS, runner.getExitCode());
+    }
+
+    @Test
+    public void forceRestartForJobThatIsAlreadyInTheRepository() throws Exception {
```

Comment:
great tests!

Reply:
BTW, now that we are here, should we do resumability checks? I saw the simple test steps that only log stuff, so we could add another ones based on reader+writer. Do you remember if we did this in another project?

Reply:
Agreed that will be handled in [EVA-1932](https://www.ebi.ac.uk/panda/jira/browse/EVA-1932)